### PR TITLE
DRDT-89 Ad Stack: Bounce Exchange

### PR DESCRIPTION
**Pull request by subinviswanathan**
[DRDT-89](https://readersdigest.atlassian.net/browse/DRDT-89)

### CODE REVIEW

File: web/wp-content/plugins/tmbi-bx/inc/cpt-bx.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: P7111439
+ * Date: 9/13/2018
+ * Time: 1:09 PM
+ */
+
+class CPT_BX extends BX_Base {
```
Comment (subinviswanathan) [inline web/wp-content/plugins/tmbi-bx/inc/cpt-bx.php]: this can be added from CMS

**Comment by fariasf** [comment]
Created new staging environment for pr-72. Here are some useful links:

[WordPress Admin](https://pr-72-drdt.pantheonsite.io/wp/wp-admin)

[Homepage](https://pr-72-drdt.pantheonsite.io/)
[Article](https://pr-72-drdt.pantheonsite.io/jobsite-tips/abandoned-cable-removal/)
[Listicle](https://pr-72-drdt.pantheonsite.io/jobsite-tips/exterior/12-scaffolding-safety-tips-and-handy-hints/)
[Page](https://pr-72-drdt.pantheonsite.io/about-construction-pro-tips/)
[Hub](https://pr-72-drdt.pantheonsite.io/jobsite-tips/)

**Comment by fariasf** [comment]
There are visual regressions! 

View the [full visual regression test report](https://circleci.com/gh/fariasf/drdt-test/2370/artifacts/0/artifacts/backstop_data/html_report/index.html)

**Comment by fariasf** [comment]
Congrats! Frontend performance tests passed!

Full Lighthouse performance results:

| Page/Branch | pr-72 | master | Result |
| --- | ---: | ---: | :---: |
| Homepage | [92](https://circleci.com/gh/fariasf/drdt-test/2369/artifacts/0/artifacts/lighthouse_results/pr-72/homepage/lighthouse.report.html) | [96](https://circleci.com/gh/fariasf/drdt-test/2369/artifacts/0/artifacts/lighthouse_results/master/homepage/lighthouse.report.html) | ✔ |
| Article | [98](https://circleci.com/gh/fariasf/drdt-test/2369/artifacts/0/artifacts/lighthouse_results/pr-72/article/lighthouse.report.html) | [98](https://circleci.com/gh/fariasf/drdt-test/2369/artifacts/0/artifacts/lighthouse_results/master/article/lighthouse.report.html) | ✔ |
| Listicle | [98](https://circleci.com/gh/fariasf/drdt-test/2369/artifacts/0/artifacts/lighthouse_results/pr-72/listicle/lighthouse.report.html) | [98](https://circleci.com/gh/fariasf/drdt-test/2369/artifacts/0/artifacts/lighthouse_results/master/listicle/lighthouse.report.html) | ✔ |
| Page | [98](https://circleci.com/gh/fariasf/drdt-test/2369/artifacts/0/artifacts/lighthouse_results/pr-72/page/lighthouse.report.html) | [98](https://circleci.com/gh/fariasf/drdt-test/2369/artifacts/0/artifacts/lighthouse_results/master/page/lighthouse.report.html) | ✔ |
| Hub | [97](https://circleci.com/gh/fariasf/drdt-test/2369/artifacts/0/artifacts/lighthouse_results/pr-72/hub/lighthouse.report.html) | [94](https://circleci.com/gh/fariasf/drdt-test/2369/artifacts/0/artifacts/lighthouse_results/master/hub/lighthouse.report.html) | ✔ |


(**Note: Tests are against the _NOADS_ version of the pages!**)

File: web/wp-content/plugins/tmbi-bx/plugin.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+*/
+
+require( 'inc/cpt-bx.php' );
+
+class BX_Controller extends BX_Base  {
+	const VERSION     = '1.2';
+	public $bx;
+
+	public function __construct() {
+		$vb = new Variant_Base();
```
Comment (fariasf) [inline web/wp-content/plugins/tmbi-bx/plugin.php]: This class is missing from the repo.

File: web/wp-content/plugins/tmbi-bx/plugin.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+	SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+	CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+	OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+	OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+require( 'inc/cpt-bx.php' );
+
+class BX_Controller extends BX_Base  {
```
Comment (fariasf) [inline web/wp-content/plugins/tmbi-bx/plugin.php]: This class is being used for namespacing only. There's no need for it.

File: web/wp-content/plugins/tmbi-bx/plugin.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+			$this->render_bx_is_blocked();
+		}
+	}
+
+	public function bx_loader() {
+		$this->bx = CPT_BX::get_instance();
+	}
+}
+
+$bxc = BX_Controller::get_instance();
```
Comment (fariasf) [inline web/wp-content/plugins/tmbi-bx/plugin.php]: `get_instance` method will be missing as `WP_Base` is no longer part of the repo.

File: web/wp-content/plugins/tmbi-bx/inc/cpt-bx.php [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+	const SCRIPT_ID = 3125;
```
Comment (fariasf) [inline web/wp-content/plugins/tmbi-bx/inc/cpt-bx.php]: This should be a configuration.

File: web/wp-content/plugins/tmbi-bx/inc/bx-base.php [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+	public function render_bx_load_failure() {
+		print( '<!-- bounce exchange script has failed to load -->' . PHP_EOL );
+	}
+
+	public function render_bx_is_blocked() {
+		print( '<!-- Bounce Exchange has been blocked -->' . PHP_EOL );
+	}
+
+	public function init() {
```
Comment (fariasf) [inline web/wp-content/plugins/tmbi-bx/inc/bx-base.php]: When does this run?

File: web/wp-content/plugins/tmbi-bx/inc/bx-base.php [hunk 7]
```diff
@@ -0,0 +1,49 @@
+<?php
+
+/**
+ * Created by PhpStorm.
+ * User: MJ
+ * Date: 3/28/2017
+ * Time: 1:10 PM
+ */
+class BX_Base {
```
Comment (fariasf) [inline web/wp-content/plugins/tmbi-bx/inc/bx-base.php]: This class is being used for namespacing only, there's no need for it.

File: web/wp-content/plugins/tmbi-bx/inc/bx-base.php [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+	const FILE_SPEC         = __DIR__;
+	const LOCALIZED_LABEL   = 'rd_bx';
+	const LOADER_SCRIPT     = 'js/smart-tag.js';
+	const LOADER_LABEL      = 'fhm-bx-loader';
```
Comment (fariasf) [inline web/wp-content/plugins/tmbi-bx/inc/bx-base.php]: Why `fhm` here?

File: web/wp-content/plugins/tmbi-bx/inc/bx-base.php [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+	public function init() {
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+	}
+
+	public function enqueue_scripts() {
+		wp_register_script(
+			static::LOADER_LABEL,
+			self::get_asset_url( static::LOADER_SCRIPT ),
+			static::DEPENDS,
+			static::$version,
```
Comment (fariasf) [inline web/wp-content/plugins/tmbi-bx/inc/bx-base.php]: What's the point of using a public static variable, that is filled from a setter (`set_version`), that's only called with a constant param (`$this->bx->set_version( self::VERSION );`)? That's a lot of code running just to pass the script version to the tag.

File: web/wp-content/plugins/tmbi-bx/inc/bx-base.php [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+	}
+
+	public function init() {
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+	}
+
+	public function enqueue_scripts() {
+		wp_register_script(
+			static::LOADER_LABEL,
+			self::get_asset_url( static::LOADER_SCRIPT ),
```
Comment (fariasf) [inline web/wp-content/plugins/tmbi-bx/inc/bx-base.php]: `get_asset_url` function is missing

**Comment by fariasf** [comment]
Updated staging environment for pr-72. Here are some useful links:

[WordPress Admin](https://pr-72-drdt.pantheonsite.io/wp/wp-admin)

[Homepage](https://pr-72-drdt.pantheonsite.io/)
[Article](https://pr-72-drdt.pantheonsite.io/jobsite-tips/abandoned-cable-removal/)
[Listicle](https://pr-72-drdt.pantheonsite.io/jobsite-tips/exterior/12-scaffolding-safety-tips-and-handy-hints/)
[Page](https://pr-72-drdt.pantheonsite.io/about-construction-pro-tips/)
[Hub](https://pr-72-drdt.pantheonsite.io/jobsite-tips/)

**Comment by fariasf** [comment]
There are visual regressions! 

View the [full visual regression test report](https://circleci.com/gh/fariasf/drdt-test/2517/artifacts/0/artifacts/backstop_data/html_report/index.html)

**Comment by fariasf** [comment]
Congrats! Frontend performance tests passed!

Full Lighthouse performance results:

| Page/Branch | pr-72 | master | Result |
| --- | ---: | ---: | :---: |
| Homepage | [95](https://circleci.com/gh/fariasf/drdt-test/2518/artifacts/0/artifacts/lighthouse_results/pr-72/homepage/lighthouse.report.html) | [96](https://circleci.com/gh/fariasf/drdt-test/2518/artifacts/0/artifacts/lighthouse_results/master/homepage/lighthouse.report.html) | ✔ |
| Article | [98](https://circleci.com/gh/fariasf/drdt-test/2518/artifacts/0/artifacts/lighthouse_results/pr-72/article/lighthouse.report.html) | [98](https://circleci.com/gh/fariasf/drdt-test/2518/artifacts/0/artifacts/lighthouse_results/master/article/lighthouse.report.html) | ✔ |
| Listicle | [98](https://circleci.com/gh/fariasf/drdt-test/2518/artifacts/0/artifacts/lighthouse_results/pr-72/listicle/lighthouse.report.html) | [98](https://circleci.com/gh/fariasf/drdt-test/2518/artifacts/0/artifacts/lighthouse_results/master/listicle/lighthouse.report.html) | ✔ |
| Page | [98](https://circleci.com/gh/fariasf/drdt-test/2518/artifacts/0/artifacts/lighthouse_results/pr-72/page/lighthouse.report.html) | [98](https://circleci.com/gh/fariasf/drdt-test/2518/artifacts/0/artifacts/lighthouse_results/master/page/lighthouse.report.html) | ✔ |
| Hub | [97](https://circleci.com/gh/fariasf/drdt-test/2518/artifacts/0/artifacts/lighthouse_results/pr-72/hub/lighthouse.report.html) | [95](https://circleci.com/gh/fariasf/drdt-test/2518/artifacts/0/artifacts/lighthouse_results/master/hub/lighthouse.report.html) | ✔ |


(**Note: Tests are against the _NOADS_ version of the pages!**)

**Comment by fariasf** [comment]
Updated staging environment for pr-72. Here are some useful links:

[WordPress Admin](https://pr-72-drdt.pantheonsite.io/wp/wp-admin)

[Homepage](https://pr-72-drdt.pantheonsite.io/)
[Article](https://pr-72-drdt.pantheonsite.io/jobsite-tips/abandoned-cable-removal/)
[Listicle](https://pr-72-drdt.pantheonsite.io/jobsite-tips/exterior/12-scaffolding-safety-tips-and-handy-hints/)
[Page](https://pr-72-drdt.pantheonsite.io/about-construction-pro-tips/)
[Hub](https://pr-72-drdt.pantheonsite.io/jobsite-tips/)

**Comment by fariasf** [comment]
There are visual regressions! 

View the [full visual regression test report](https://circleci.com/gh/fariasf/drdt-test/2525/artifacts/0/artifacts/backstop_data/html_report/index.html)

**Comment by fariasf** [comment]
Congrats! Frontend performance tests passed!

Full Lighthouse performance results:

| Page/Branch | pr-72 | master | Result |
| --- | ---: | ---: | :---: |
| Homepage | [95](https://circleci.com/gh/fariasf/drdt-test/2524/artifacts/0/artifacts/lighthouse_results/pr-72/homepage/lighthouse.report.html) | [96](https://circleci.com/gh/fariasf/drdt-test/2524/artifacts/0/artifacts/lighthouse_results/master/homepage/lighthouse.report.html) | ✔ |
| Article | [98](https://circleci.com/gh/fariasf/drdt-test/2524/artifacts/0/artifacts/lighthouse_results/pr-72/article/lighthouse.report.html) | [97](https://circleci.com/gh/fariasf/drdt-test/2524/artifacts/0/artifacts/lighthouse_results/master/article/lighthouse.report.html) | ✔ |
| Listicle | [98](https://circleci.com/gh/fariasf/drdt-test/2524/artifacts/0/artifacts/lighthouse_results/pr-72/listicle/lighthouse.report.html) | [97](https://circleci.com/gh/fariasf/drdt-test/2524/artifacts/0/artifacts/lighthouse_results/master/listicle/lighthouse.report.html) | ✔ |
| Page | [98](https://circleci.com/gh/fariasf/drdt-test/2524/artifacts/0/artifacts/lighthouse_results/pr-72/page/lighthouse.report.html) | [98](https://circleci.com/gh/fariasf/drdt-test/2524/artifacts/0/artifacts/lighthouse_results/master/page/lighthouse.report.html) | ✔ |
| Hub | [97](https://circleci.com/gh/fariasf/drdt-test/2524/artifacts/0/artifacts/lighthouse_results/pr-72/hub/lighthouse.report.html) | [92](https://circleci.com/gh/fariasf/drdt-test/2524/artifacts/0/artifacts/lighthouse_results/master/hub/lighthouse.report.html) | ✔ |


(**Note: Tests are against the _NOADS_ version of the pages!**)

**Comment by subinviswanathan** [comment]
[DRDT-89](https://readersdigest.atlassian.net/browse/DRDT-89)

**Comment by fariasf** [comment]
Updated staging environment for pr-72. Here are some useful links:

[WordPress Admin](https://pr-72-drdt.pantheonsite.io/wp/wp-admin)

[Homepage](https://pr-72-drdt.pantheonsite.io/)
[Article](https://pr-72-drdt.pantheonsite.io/jobsite-tips/abandoned-cable-removal/)
[Listicle](https://pr-72-drdt.pantheonsite.io/jobsite-tips/exterior/12-scaffolding-safety-tips-and-handy-hints/)
[Page](https://pr-72-drdt.pantheonsite.io/about-construction-pro-tips/)
[Hub](https://pr-72-drdt.pantheonsite.io/jobsite-tips/)

**Comment by fariasf** [comment]
There are visual regressions! 

View the [full visual regression test report](https://circleci.com/gh/fariasf/drdt-test/2539/artifacts/0/artifacts/backstop_data/html_report/index.html)

**Comment by fariasf** [comment]
Congrats! Frontend performance tests passed!

Full Lighthouse performance results:

| Page/Branch | pr-72 | master | Result |
| --- | ---: | ---: | :---: |
| Homepage | [94](https://circleci.com/gh/fariasf/drdt-test/2538/artifacts/0/artifacts/lighthouse_results/pr-72/homepage/lighthouse.report.html) | [97](https://circleci.com/gh/fariasf/drdt-test/2538/artifacts/0/artifacts/lighthouse_results/master/homepage/lighthouse.report.html) | ✔ |
| Article | [98](https://circleci.com/gh/fariasf/drdt-test/2538/artifacts/0/artifacts/lighthouse_results/pr-72/article/lighthouse.report.html) | [98](https://circleci.com/gh/fariasf/drdt-test/2538/artifacts/0/artifacts/lighthouse_results/master/article/lighthouse.report.html) | ✔ |
| Listicle | [98](https://circleci.com/gh/fariasf/drdt-test/2538/artifacts/0/artifacts/lighthouse_results/pr-72/listicle/lighthouse.report.html) | [98](https://circleci.com/gh/fariasf/drdt-test/2538/artifacts/0/artifacts/lighthouse_results/master/listicle/lighthouse.report.html) | ✔ |
| Page | [98](https://circleci.com/gh/fariasf/drdt-test/2538/artifacts/0/artifacts/lighthouse_results/pr-72/page/lighthouse.report.html) | [98](https://circleci.com/gh/fariasf/drdt-test/2538/artifacts/0/artifacts/lighthouse_results/master/page/lighthouse.report.html) | ✔ |
| Hub | [97](https://circleci.com/gh/fariasf/drdt-test/2538/artifacts/0/artifacts/lighthouse_results/pr-72/hub/lighthouse.report.html) | [97](https://circleci.com/gh/fariasf/drdt-test/2538/artifacts/0/artifacts/lighthouse_results/master/hub/lighthouse.report.html) | ✔ |


(**Note: Tests are against the _NOADS_ version of the pages!**)

**Comment by fariasf** [comment]
Updated staging environment for pr-72. Here are some useful links:

[WordPress Admin](https://pr-72-drdt.pantheonsite.io/wp/wp-admin)

[Homepage](https://pr-72-drdt.pantheonsite.io/)
[Article](https://pr-72-drdt.pantheonsite.io/jobsite-tips/abandoned-cable-removal/)
[Listicle](https://pr-72-drdt.pantheonsite.io/jobsite-tips/exterior/12-scaffolding-safety-tips-and-handy-hints/)
[Page](https://pr-72-drdt.pantheonsite.io/about-construction-pro-tips/)
[Hub](https://pr-72-drdt.pantheonsite.io/jobsite-tips/)

**Comment by fariasf** [comment]
There are visual regressions! 

View the [full visual regression test report](https://circleci.com/gh/fariasf/drdt-test/2723/artifacts/0/artifacts/backstop_data/html_report/index.html)

**Comment by fariasf** [comment]
Congrats! Frontend performance tests passed!

Full Lighthouse performance results:

| Page/Branch | pr-72 | master | Result |
| --- | ---: | ---: | :---: |
| Homepage | [93](https://circleci.com/gh/fariasf/drdt-test/2722/artifacts/0/artifacts/lighthouse_results/pr-72/homepage/lighthouse.report.html) | [94](https://circleci.com/gh/fariasf/drdt-test/2722/artifacts/0/artifacts/lighthouse_results/master/homepage/lighthouse.report.html) | ✔ |
| Article | [98](https://circleci.com/gh/fariasf/drdt-test/2722/artifacts/0/artifacts/lighthouse_results/pr-72/article/lighthouse.report.html) | [98](https://circleci.com/gh/fariasf/drdt-test/2722/artifacts/0/artifacts/lighthouse_results/master/article/lighthouse.report.html) | ✔ |
| Listicle | [97](https://circleci.com/gh/fariasf/drdt-test/2722/artifacts/0/artifacts/lighthouse_results/pr-72/listicle/lighthouse.report.html) | [97](https://circleci.com/gh/fariasf/drdt-test/2722/artifacts/0/artifacts/lighthouse_results/master/listicle/lighthouse.report.html) | ✔ |
| Page | [98](https://circleci.com/gh/fariasf/drdt-test/2722/artifacts/0/artifacts/lighthouse_results/pr-72/page/lighthouse.report.html) | [98](https://circleci.com/gh/fariasf/drdt-test/2722/artifacts/0/artifacts/lighthouse_results/master/page/lighthouse.report.html) | ✔ |
| Hub | [97](https://circleci.com/gh/fariasf/drdt-test/2722/artifacts/0/artifacts/lighthouse_results/pr-72/hub/lighthouse.report.html) | [95](https://circleci.com/gh/fariasf/drdt-test/2722/artifacts/0/artifacts/lighthouse_results/master/hub/lighthouse.report.html) | ✔ |


(**Note: Tests are against the _NOADS_ version of the pages!**)

**Pull request merged**